dvc/command/add.py

```diff
@@ -57,6 +58,12 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Allow targets that are outside of the DVC repository.",
     )
+    parser.add_argument(
+        "--glob",
+        action="store_true",
+        default=False,
+        help="Allows targets containing shell-style wildcards.",
+    )
```

Comment:
Hi! A few questions (related to documenting this per iterative/dvc.org/issues/1928):

- What does "glob" abbreviate or stand for?
- Why is a flag needed to allow/enable wildcards?
- Is there a short description of what "shell-style" means exactly? Or a link to some Linux doc, Python package, or something specifying what can be done in terms of wildcards?

Reply:
I think it's more natural to use wildcards in the default mode without any flag, just like what we do in Unix. And it‘s better to extract the `glob` to some common function that could be used in all commands.

Reply:
@karajan1001 I personally wasn't ready to enable this by default right away as I'm not sure it won't affect negatively some existing usecases :slightly_frowning_face:  Once we're ready for it, it will be as simple as turning that flag on by default. For PS users `--glob` will be a temporary inconvenience, but at least the functionality is being slowly filled up.

@jorgeorpinel Maybe just forget about the doc ticket for it for now, we'll be adding more stuff for it in the future.

> What does "glob" abbreviate or stand for?

https://docs.python.org/3/library/glob.html 

> Why is a flag needed to allow/enable wildcards?

Usually, your shell handles the expansion and it will still do that. Our --glob is useful for programatic use or in shells that don't support explansion natively (e.g. powershell). I'm personally not ready to enable by default right now.

> Is there a short description of what "shell-style" means exactly? Or a link to some Linux doc, Python package, or something specifying what can be done in terms of wildcards?

https://docs.python.org/3/library/glob.html

Reply:
Cool, just wanted to understand this. The initial docs PR is ready for merge.